### DISCUSSION TITLE
support webpack5 new library type 'module'

### DISCUSSION THREAD

**Discussion by cncolder** [webpack-contrib/webpack-bundle-analyzer#587] [Ideas]
#### Issue description

Currently, It cannot parse properly when webpack5 config library type to `module`. Because webpack5 append some codes like `export {a, b, c}` to the end of the bundle.

```ts
const config: webpack.Configuration = {
  output: {
    libraryTarget: 'module',
  },
  experiments: {
    outputModule: true,
  },
  plugins: [new BundleAnalyzerPlugin({ analyzerMode: 'static' })],
}
```

#### Technical info

<pre>
  System:
    OS: macOS 11.2.3
    CPU: (4) x64 Intel(R) Core(TM) i5-7500 CPU @ 3.40GHz
    Memory: 3.01 GB / 24.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 15.11.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.6.0 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  npmPackages:
    case-sensitive-paths-webpack-plugin: 2.4.0 => 2.4.0 
    fork-ts-checker-webpack-plugin: 6.2.0 => 6.2.0 
    webpack: 5.28.0 => 5.28.0 
    webpack-bundle-analyzer: 4.4.0 => 4.4.0 
    webpack-cli: 4.5.0 => 4.5.0 
    webpack-merge: 5.7.3 => 5.7.3 
</pre>

#### Debug info
How do you use this module? As CLI utility or as plugin?

plugin

If CLI, what command was used? (e.g. `webpack-bundle-analyzer -O path/to/stats.json`)

If plugin, what options were provided? (e.g. `new BundleAnalyzerPlugin({ analyzerMode: 'disabled', generateStatsFile: true })`)

`new BundleAnalyzerPlugin({ analyzerMode: 'static' })`

What other Webpack plugins were used?

**Comment by valscion** [comment]
Thanks for opening an issue! PR is welcome ☺️

**Comment by g3tr1ght** [comment]
https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/d96ebed7d0dd38e6a0b9a935aaac0d3c24722944/src/parseUtils.js#L13

**Comment by kiranbayram** [comment]
@valscion Do you have any pointers how this can be implemented? I'd try to help if i can

**Comment by Airkro** [comment]
Turn

https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/d96ebed7d0dd38e6a0b9a935aaac0d3c24722944/src/parseUtils.js#L13

To

```js
sourceType: 'module'
```

will work

**Comment by opeologist** [comment]
any update on this? would love to analyze these types of bundles

**Comment by ml1nk** [comment]
I tried changing the sourceType but it didn't work. I think the data structure changes somewhat.

**Comment by dbrenot-pelmorex** [comment]
Heads up that this is still an issue.

**Comment by valscion** [comment]
Anyone who wants to can take a stab at fixing this with a PR ☺️

**Comment by rcherny** [comment]
Just as additional information, the proposed fix:

```js
sourceType: 'module'
```
Resolved this for me.

**Reply by eamodio** [reply]
Same for me /cc @valscion

**Reply by webholics** [reply]
Also works for me. But is there an update planned?

**Reply by alexander-akait** [reply]
Feel free to send a pr or open an issue

**Reply by eamodio** [reply]
@alexander-akait I can send a PR that just switches to 'module', but I don't know what the other effects would be.

**Reply by alexander-akait** [reply]
I think we can get this value from configuration and switch safely

**Reply by eamodio** [reply]
I've opened https://github.com/webpack-contrib/webpack-bundle-analyzer/pull/649
